forte/pipeline.py

```diff
@@ -281,24 +319,63 @@ def _dump_to_config(self):
         Returns:
             dict: A dictionary storing IR.
         """
-        configs: List[Dict] = []
-        configs.append(
+        configs: Dict = {
+            "forte_ir_version": self.FORTE_IR_VERSION,
+            "components": list(),
+            "states": dict(),
+        }
+
+        # Serialize pipeline components
+        configs["components"].append(
             {
                 "type": ".".join(
                     [self._reader.__module__, type(self._reader).__name__]
                 ),
                 "configs": self._reader_config.todict(),
             }
         )
-        for component, config in zip(self.components, self.component_configs):
-            configs.append(
+        for component, config, selector in zip(
+            self.components, self.component_configs, self._selectors
+        ):
+            configs["components"].append(
                 {
                     "type": ".".join(
                         [component.__module__, type(component).__name__]
                     ),
                     "configs": config.todict(),
+                    "selector": {
+                        "type": ".".join(
+                            [selector.__module__, type(selector).__name__]
+                        ),
+                        # TODO: This presumes that class attributes' names are
```

Comment:
One alternative is we don't store `kwargs`, but instead assign attributes using `setattr`. But we might run into error during `__init__` (for example, if the selector has a `__init__` time kwargs check).

Actually, a more important problem is that the `selectors` are currently having arbitrary init parameters. Some of them may not be serializable.